include/swoole.h

```diff
@@ -555,26 +579,10 @@ typedef struct _swConnection
     swString ssl_client_cert;
 #endif
     sw_atomic_t lock;
-} swConnection;
-
-typedef struct _swProtocol
-{
-    /* one package: eof check */
-    uint8_t split_by_eof;
-    uint8_t package_eof_len;  //数据缓存结束符长度
-    char package_eof[SW_DATA_EOF_MAXLEN + 1];  //数据缓存结束符
-
-    char package_length_type;  //length field type
-    uint8_t package_length_size;
-    uint16_t package_length_offset;  //第几个字节开始表示长度
-    uint16_t package_body_offset;  //第几个字节开始计算长度
-    uint32_t package_max_length;
-
-    void *private_data;
+    swProtocol protocol;
+    int (*onRead)(swReactor *reactor, struct _swConnection *conn, swEvent *event);
```

Comment:
now, protocol and onRead function has been added to connection, which make it possible to support different protocol on the same port.